doc/specs/Proto extensions-spec.md

```diff
@@ -38,66 +38,178 @@ in between 2 and 3:
-A straightforward way to do this would be by providing the name of the profile. Since this modification would
-occur at the step _before_ any user changes to names are layered, the names of the profiles are consistent and
-can be used for identification. For example, the name of a profile generated by the WSL profile generator is
-simply the argument to "wsl.exe -d \<name\>", the name of a profile generated by the Powershell profile generator
-would be "Powershell Core" or "Powershell (ARM)" etc depending on the version.
-
-We might run into the case where multiple json stubs modify the same profile and so they override each other. We
-will need some sort of hierarchy to determine which stub to ultimately use. My initial thought is that we could
-check the owner of the json file and see if it matches the profile they are trying to modify (so something like
-only Powershell can modify the Powershell profiles and only Ubuntu can modify the Ubuntu profile etc). Independent
-parties will need to create full profiles with the desired commandline argument instead.
+Note that currently, we generate a GUID for dynamic profiles using the "initial" name of the profile (i.e. before
+any user changes are applied). For example, the "initial" name of a WSL profile is the \<name\> argument to
+"wsl.exe -d \<name\>", and the "initial" name of a Powershell profile is something like "Powershell (ARM)" - depending
+on the version. Thus, the stub creator could simply use the same uuidv5 GUID generator we do on that name to obtain the
+GUID. Then, in the stub they provide the GUID can be used to identify which profile to modify.
+
+Naturally, we will have to provide documentation on how they could generate the desired GUID themselves. In that documentation,
+we will also provide examples showing how the current GUIDs are generated for clarity's sake. 
+
+We might run into the case where multiple json stubs modify the same profile and so they override each other. For the initial implementation, we
+are simply going to apply _all_ the changes. Eventually, we will probably want some sort of hierarchy to determine
+an order to which changes are applied.
+
+Here is an example of a json stub that modifies an existing profile (specifically the Azure cloud shell profile):
+
+```js
+{
+    "guid": "{b453ae62-4e3d-5e58-b989-0a998ec441b8}",
+    "fontSize": 16,
+    "fontWeight": "thin"
+}
+```
+
+**NOTE**: This will *not* change the way the profile looks in the user's settings file. The Azure cloud shell profile
+in their settings file (assuming they have made no changes) will still be as below. 
+
+```js
+{
+    "guid": "{b453ae62-4e3d-5e58-b989-0a998ec441b8}",
+    "hidden": false,
+    "name": "Azure Cloud Shell",
+    "source": "Windows.Terminal.Azure"
+}
+```
+However, the user is free to make changes to it as usual and those changes will have the 'final say'. 

```

Comment:
In this example, if the user has "fontSize" defined in profiles.defaults, which one is applied? (might be worth adding it to the numbered list earlier too).

Reply:
The user's one! Its still considered a 'user setting' so it will get applied last